### DISCUSSION TITLE
How to avoid encoding variable framerate?

### DISCUSSION THREAD

**Discussion by mpek** [RandomEngy/VidCoder#1297] [General]
This is a little awkward, because everything was set as necessary (constant framerate) and still variable framerate is shown in Mediainfo. 
According to Mediainfo, a rendered video is encoded in Variable framerate, even though Constant framerate is set in preset. 

Source is MKV with NVEnc HEVC CQ0, six audio streams (3 DTS and 3 AAC), 5 subtitles (SSA). The source file is made by VidCoder. 

Source file MediaInfo:
Frame rate mode                : Constant
Frame rate                     : 23.976 (24000/1001) FPS
Color space                    : YUV

Destination is:
- MP4 (ticked: include chapter markers, web optimized)
- HEVC (no cropping, no video filters, H.265 NVEnc, Framerate: Same as source, Constant framerate, Target size is 1910 MB)
- Allow passthrough for: AAC
- Take all subtitles over to new file and burn in one of them. 

VidCoder has to encode just the video with the fitting bitrate and combine all the streams into a MP4. 

Everything is in the new file, but the video is now in Variable framerate and it should be in Constant framerate. 

Encoded whole video MediaInfo: 
Frame rate mode                : Variable
Frame rate                     : 23.976 (24000/1001) FPS
Minimum frame rate             : 15.947 FPS
Maximum frame rate             : 23.976 FPS

Note: If just the first chapter is being encoded, from the same video, the framerate shows as Constant framerate. The video has 18 chapters. 

Encoded 1 Chapter MediaInfo:
Frame rate mode                : Constant
Frame rate                     : 23.976 (24000/1001) FPS
Color space                    : YUV

When encoding the first 9 chapters the resulting framerate is Variable!
When encoding the first 4 chapters the resulting framerate is Variable!
When encoding the first 3 chapters the resulting framerate is Variable!
When encoding the first 2 chapters the resulting framerate is Variable!

Could you pleae tell me how to get the whole video encoded in constant framerate?

I know that you cannot test with the same source file, but I can repeat this issue with other files that VidCoder/Handbrake encoded with source files in NVEnc HEVC, because this is the main codec i use with my collection. Could this be caused, because of reencoding HEVC? But if, why?

**Comment by RandomEngy** [comment]
If you choose Constant Framerate, VidCoder will apply that filter to the encode and you should see this in the log:

```
[10:03:17]      + Framerate Shaper (mode=1)
[10:03:17]        + frame rate: 23.976 fps -> constant 23.976 fps
```

If you see that, but it's still variable frame rate output, that means the issue is in the HandBrake engine.
